src/coins.cpp

```diff
@@ -679,7 +679,7 @@ void CCoinsViewCache::SetNullifiers(const CTransaction& tx, bool spent) {
         }
     }
     for (const auto& spendDescription : tx.GetSaplingSpends()) {
-        std::pair<CNullifiersMap::iterator, bool> ret = cacheSaplingNullifiers.insert(std::make_pair(spendDescription.nullifier(), CNullifiersCacheEntry()));
+        std::pair<CNullifiersMap::iterator, bool> ret = cacheSaplingNullifiers.insert(std::make_pair(uint256::FromRawBytes(spendDescription.nullifier()), CNullifiersCacheEntry()));
```

Comment:
Per our conversation in our pairing, this is not converting to/from bytes, because a nullifier _is_ bytes. What this is doing is converting on the C++ side between the C++ array type `std::array<uint8_t, 32>` (which is used by `cxx` to map to the Rust array `[u8; 32]`), and the C array type `uint8_t[32]` inside `uint256` that is used for nullifiers for legacy reasons. On the rust side, we're "parsing" to a `Nullifier`, which is just a newtype around `[u8; 32]`.